src/backend/columnar/columnar_customscan.c

```diff
@@ -274,8 +281,14 @@ ColumnarSetRelPathlistHook(PlannerInfo *root, RelOptInfo *rel, Index rti,
 							errmsg("sample scans not supported on columnar tables")));
 		}
 
-		/* columnar doesn't support parallel paths */
-		rel->partial_pathlist = NIL;
+		if (list_length(rel->partial_pathlist) != 0)
+		{
+			/*
+			 * Parallel scans on columnar tables are already discardad by
+			 * ColumnarGetRelationInfoHook but be on the safe side.
+			 */
+			ereport(ERROR, (errmsg("parallel scans on columnar are not supported")));
+		}
```

Comment:
you mentioned this code is unreachable. IIRC Postgres uses `elog(ERROR, ...)` instead of `ereport(ERROR, ...)` for these situations.